src/frequenz/client/electricity_trading/_client.py

```diff
@@ -49,6 +50,25 @@ class _Sentinel:
 NO_VALUE = _Sentinel()
 
 
+def validate_decimal_places(value: Decimal, decimal_places: int, name: str) -> None:
+    """
+    Validate that the decimal places of a given value do not exceed a specified limit.
+
+    Args:
+        value: The value to be checked.
+        decimal_places: The maximum allowed decimal places.
+        name: The name of the value (for error messages).
+
+    Raises:
+        ValueError: If the value has more decimal places than allowed.
+    """
+    if value is not None:
+        if value.as_tuple().exponent < -decimal_places:  # type: ignore
+            raise ValueError(
+                f"The {name} cannot have more than {decimal_places} decimal places."
+            )
```

Comment:
I'd suggest to make this function more bullet-proof, something around this (haven't tested it)
```python
    assert decimal_places >= 0, "The decimal places must be a non-negative integer."

    try:
        exponent = value.as_tuple().exponent
        if abs(int(exponent)) > decimal_places:
            raise ValueError(
                f"The {name} cannot have more than {decimal_places} decimal places."
            )
    except InvalidOperation as exc:
        raise ValueError(f"The {name} is not a valid decimal number.") from exc
```